mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -1161,6 +1186,30 @@ void mav_vehicle::update()
             }
         }
     }
+
+    // Check if autorotation is active in order to request rotation if needed,
+    // if not already rotating. Need also to make sure that a mission waypoint
+    // is not currently being sent.
+    if (this->waypoint_autorotate &&
+        (this->mstatus == mission_status::NORMAL ||
+         this->mstatus == mission_status::DETOURING) &&
+        !this->sending_mission) {
```

Comment:
You could go with early return here, but it's also fine to keep it like this (in case you wanna add more code to this function later)